### PR TITLE
Add link fix script

### PR DESCRIPTION
**Pull request by queengooborg**
This PR adds a brand new fix script that I'm proud to introduce!  Following up to my refactoring work for the links tests (#5187), I've put together a fix script to automatically fix issues with links that we lint for.  This takes the errors that come from the linter and performs replacement of the links based upon those errors.

### CODE REVIEW

**Review by alshamiri5 — APPROVED** [review]

**Comment by foolip** [comment]
@queengooborg can you rebase and give an example for how to test that this is working?

**Comment by queengooborg** [comment]
Of course -- one easy test to perform is to change any MDN URL or bug link to `http://` (rather than `https://`) as it is now.

File: scripts/fix-links.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const fs = require('fs');
+const { platform } = require('os');
+
+const { processData } = require('../test/linter/test-links.js');
+
+/** Determines if the OS is Windows */
+const IS_WINDOWS = platform() === 'win32';
+
+/**
+ * @param {Promise<void>} filename
```
Comment (caugner) [inline scripts/fix-links.js:16]: Based on the following, filename is a `string`:

https://github.com/mdn/browser-compat-data/blob/1d4f83058631578fd298b759cfc7ae1cb90c9655/test/linter/test-links.js#L11-L17

```suggestion
 * @param {string} filename
```

File: scripts/fix-links.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+const fixLinks = (filename) => {
+  let errors = processData(filename);
+  let original = fs.readFileSync(filename, 'utf-8').trim();
```
Comment (caugner) [inline scripts/fix-links.js:20]: Guess ESLint would complain about this:

```suggestion
  const errors = processData(filename);
  const original = fs.readFileSync(filename, 'utf-8').trim();
```

File: scripts/fix-links.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  let data = original;
+
+  if (IS_WINDOWS) {
+    // prevent false positives from git.core.autocrlf on Windows
+    data = data.replace(/\r/g, '');
+  }
+
+  for (let error of errors) {
```
Comment (caugner) [inline scripts/fix-links.js:28]: ditto
```suggestion
  for (const error of errors) {
```

**Review by caugner — CHANGES_REQUESTED** [review]
LGTM and works (`npm run fix` after changing `http://` to `https://` in `mdn_url`s or adding `/en-US/` to a url).

Just few nits.

**Pull request merged**